packages/x-date-pickers/src/PickersSectionList/PickersSectionList.tsx

```diff
@@ -94,10 +94,28 @@ function PickersSection(props: PickersSectionProps) {
     ownerState: { ...ownerState, separatorPosition: 'after' },
   });
 
+  // Only propagate blur when focus leaves the SectionContent entirely.
+  const sectionContentRef = React.useRef<HTMLSpanElement>(null);
+  const handleSectionContentRef = useForkRef((sectionContentProps as any).ref, sectionContentRef);
+
   return (
     <Section {...sectionProps}>
       <SectionSeparator {...sectionSeparatorBeforeProps} />
-      <SectionContent {...sectionContentProps} />
+      <SectionContent
+        {...sectionContentProps}
+        ref={handleSectionContentRef}
+        onBlur={(event) => {
```

Comment:
I think it would be easier to ready with a `handleContentBlur` method before the render
But other than that looks good 👍